santoku/utils/url_handler.py

```diff
@@ -1,205 +1,337 @@
 import ipaddress
-from typing import List
-from urllib.parse import urlparse
+from typing import List, Optional, Tuple
+from urllib.parse import ParseResult, urlparse
 
 import tldextract
 
 
-class InvalidURLError(Exception):
-    def __init__(self, message):
-        super().__init__(message)
+class URLHandler:
+    """
+    This class perfoms some parsing and transformation to a URL and provide methods to obtain
+    different forms of this URL.
 
+    A URL is split into different components: `scheme`, `subdomains`, `base_domain`, `suffix` and
+    `path`.
```

Comment:
Put a link to some official source (wiki, mozilla, etc) where this is described in detail